libs/blocks/global-navigation/utilities/menu/menu.js

```diff
@@ -276,6 +277,21 @@ const decorateColumns = async ({ content, separatorTagName = 'H5' } = {}) => {
   }
 };
 
+const decorateCrossCloudMenu = (content) => {
+  const crossCloudMenuEl = content.querySelector('.cross-cloud-menu');
+  if (!crossCloudMenuEl) return;
+
+  const crossCloudMenuContent = toFragment`<div class="feds-crossCloudMenu"></div>`;
+  crossCloudMenuContent.append(...[...crossCloudMenuEl.children].map((el, index) => {
+    const decoratedLink = decorateLinkGroup(el, index);
+    if (index === 0) decoratedLink.prepend(toFragment`${CONFIG.icons.home}`);
+    return toFragment`<div class="feds-crossCloudMenu-item">${decoratedLink}</div>`;
+  }));
+
+  content.append(toFragment`<div class="feds-crossCloudMenu-wrapper">${crossCloudMenuContent}</div>`);
+  crossCloudMenuEl.remove();
+};
```

Comment:
```suggestion
const decorateCrossCloudMenu = (content) => {
  const crossCloudMenuEl = content.querySelector('.cross-cloud-menu');
  if (!crossCloudMenuEl) return;

  const crossCloudMenu = toFragment`<div class="feds-crossCloudMenu-wrapper"<div class="feds-crossCloudMenu"></div></div>`;
  // I'm not a big fan of the double destructing assignment
  crossCloudMenu.append(...[...crossCloudMenuEl.children].map((el, index) => {
    const decoratedLink = decorateLinkGroup(el, index);
    // Does this need to be nested?
    if (index === 0) decoratedLink.prepend(toFragment`${CONFIG.icons.home}`);
    return toFragment`<div class="feds-crossCloudMenu-item">${decoratedLink}</div>`;
  }));

  content.append(crossCloudMenu);
  crossCloudMenuEl.remove();
};
```

Reply:
1. The use of 'forEach' here is a tradeoff. It makes the code slightly longer, but also makes it more readable. Made the change 👍
2. On the AEM version of the Global Navigation I saw it's nested, so I tried to keep the consistency